### PR TITLE
Fixes for #5663, #5863, #5364,  #5765, #5899, #5912

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
Fixes for followings
1. https://github.com/wso2/product-apim/issues/5899
2. https://github.com/wso2/product-apim/issues/5765
3. https://github.com/wso2/product-apim/issues/5364
4. https://github.com/wso2/product-apim/issues/5863
5. https://github.com/wso2/product-apim/issues/5663
6. https://github.com/wso2/product-apim/issues/5912
7. https://github.com/wso2/product-apim/issues/5968
7. https://github.com/wso2/product-apim/issues/5967

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java [hunk 1]
```diff
@@ -2620,7 +2620,7 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
     public void testDeleteAPI() throws RegistryException, UserStoreException, APIManagementException,
             WorkflowException {
         APIIdentifier identifier = new APIIdentifier("admin-AT-carbon.super", "API1", "1.0.0");
-
+        String apiUuid = "12345w";
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: What's this?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java [hunk 2]
```diff
@@ -2670,13 +2670,14 @@ public void testDeleteAPI() throws RegistryException, UserStoreException, APIMan
...
 
-        apiProvider.deleteAPI(identifier);
+        apiProvider.deleteAPI(identifier, apiUuid);
     }
 
     @Test(expected = APIManagementException.class)
     public void testDeleteAPI_RegistryException() throws RegistryException, UserStoreException, APIManagementException,
             WorkflowException {
         APIIdentifier identifier = new APIIdentifier("admin-AT-carbon.super", "API1", "1.0.0");
+        String apiUuid = "12345w";
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: What's this?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java [hunk 3]
```diff
@@ -2747,17 +2749,18 @@ public void testDeleteAPI_WFException() throws RegistryException, UserStoreExcep
...
 
-        apiProvider.deleteAPI(identifier);
+        apiProvider.deleteAPI(identifier, apiUuid);
     }
 
     @Test
     public void testDeleteAPI_WithActiveSubscriptions() throws RegistryException, UserStoreException,
             APIManagementException, WorkflowException {
         APIIdentifier identifier = new APIIdentifier("admin-AT-carbon.super", "API1", "1.0.0");
+        String apiUuid = "12345w";
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: What's this?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 1 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: It's a sample api_UUID.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 2 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: It's a sample api_UUID for testing purpose

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 3 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: It's a sample api_UUID for testing purpose

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 2 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: Same as above

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java (hunk 3 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: Same as above

**Pull request merged**